activation/handler.go

```diff
@@ -202,12 +205,36 @@ func (h *Handler) SyntacticallyValidateAtx(ctx context.Context, atx *types.Activ
 			return nil, fmt.Errorf("initial Post indices included in challenge does not equal to the initial Post indices included in the atx")
 		}
 
+		if atx.CommitmentATX == *types.EmptyATXID {
+			return nil, fmt.Errorf("no prevATX declared, but commitmentATX is missing")
+		}
+
+		if atx.CommitmentATX != h.goldenATXID {
+			commitmentAtx, err := h.cdb.GetAtxHeader(atx.CommitmentATX)
+			if err != nil {
+				return nil, fmt.Errorf("commitment atx not found")
+			}
+			if !atx.PubLayerID.After(commitmentAtx.PubLayerID) {
+				return nil, fmt.Errorf("atx layer (%v) must be after commitment atx layer (%v)", atx.PubLayerID, commitmentAtx.PubLayerID)
+			}
+			if d := atx.PubLayerID.Difference(commitmentAtx.PubLayerID); d > h.layersPerEpoch {
+				return nil, fmt.Errorf("expected distance of one epoch (%v layers) from commitment atx but found %v", h.layersPerEpoch, d)
+			}
+		} else {
+			publicationEpoch := atx.PublishEpoch()
+			if !publicationEpoch.NeedsGoldenPositioningATX() {
+				return nil, fmt.Errorf("golden atx used for commitment atx in epoch %d, but is only valid in epoch 1", publicationEpoch)
+			}
+		}
+
 		// Use the NIPost's Post metadata, while overriding the challenge to a zero challenge,
 		// as expected from the initial Post.
 		initialPostMetadata := *atx.NIPost.PostMetadata
 		initialPostMetadata.Challenge = shared.ZeroChallenge
-		if err := h.nipostValidator.ValidatePost(atx.NodeID().ToBytes(), atx.InitialPost, &initialPostMetadata, uint(atx.NumUnits)); err != nil {
-			return nil, fmt.Errorf("invalid initial Post: %v", err)
+
+		commitment := sha256.Sum256(append(atx.NodeID().ToBytes(), atx.CommitmentATX[:]...))
+		if err := h.nipostValidator.ValidatePost(commitment[:], atx.InitialPost, &initialPostMetadata, uint(atx.NumUnits)); err != nil {
```

Comment:
curious. why copy here?

Reply:
I'm not sure I understand? To b able to validate the post challenge I need to use the hash of the concatenation of the node ID and the commitment ATX as input.

Reply:
yes. but why make a copy at call site? isn't this an implementation details for `ValidatePost()`?
`h.nipostValidator.ValidatePost(commitment[:],...)`